secretsmanager_secret: add 'overwrite' parameter

brsolomon-deloitte opened the pull request:
##### SUMMARY

Adds an 'overwrite' parameter to secretsmanager_secret

    - If set to True, an existing secret with the same name will be overwritten.
    - If set to False, a secret with the given name will only be created if none exists.

Fixes #1626 

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

secretsmanager_secret
##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```paste below

```

tremble commented:
@brsolomon-deloitte Thanks for taking the time to submit this PR

I've rebased the PR, added some integration tests and a changelog fragment.

brsolomon-deloitte commented:
> @brsolomon-deloitte Thanks for taking the time to submit this PR
> 
> I've rebased the PR, added some integration tests and a changelog fragment.

Thanks for rounding out the rest of the PR and getting it merged! This will be a valuable feature for us.
